src/pages/index.js

```diff
@@ -6,62 +6,123 @@ import PopupWithImage from "../components/PopupWithImage.js";
+
+//API//
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "2e9b8c64-af02-41c6-9bfd-4754c20777e9",
+    "Content-type": "application/json",
+  },
+});
+
+//UserInfo//
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
 
 /* -------------------------------------------------------------------------- */
 /*                                  Elements                                  */
 /* -------------------------------------------------------------------------- */
 
-//Edit Profile
+//Edit Profile//
 const profileEditModal = document.querySelector("#profile-edit-modal");
 const profileEditForm = document.forms["profile-edit-form"];
 const profileEditBtn = document.querySelector("#profile-edit-button");
+const modalInputTitle = document.querySelector("#profile-edit-modal");
 const profileTitleInput = profileEditModal.querySelector(
   "#profile-title-input"
 );
 const profileDescriptionInput = profileEditModal.querySelector(
   "#profile-description-input"
 );
 
-//Add New Card
+//Add New Card //
 const addCardModal = document.querySelector("#add-card-modal");
 const addCardForm = document.forms["add-card-form"];
 const addCardBtn = document.querySelector(".profile__add-button");
 
-const renderCard = (cardData) => {
-  const card = createCard(cardData);
-  section.addItem(card);
-};
+function renderCard(cardData) {
+  const cardElement = createCard(cardData);
+  cardSection.addItem(cardElement);
+}
 
 /* -------------------------------------------------------------------------- */
 /*                                  Popups                                    */
 /* -------------------------------------------------------------------------- */
-//Edit Profile Popup
+//Edit Profile Popup //
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileEditSubmit
+  (profileData) => {
+    handleProfileEditSubmit;
+    editProfilePopup.renderLoading(true);
+
+    api
+      .updateProfileInfo(profileData.title, profileData.description)
+      .then((updatedUserInfo) => {
+        userInfo.setUserInfo(updatedUserInfo);
+        editProfilePopup.close();
+      })
+      .catch(console.error)
+      .finally(() => {
+        editProfilePopup.renderLoading(false);
+      });
+  }
 );
 editProfilePopup.setEventListeners();
 
-//Add New Card Popup
-const newCardPopup = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
+//Add New Card Popup //
+const newCardPopup = new PopupWithForm("#add-card-modal", ({ title, url }) => {
+  handleAddCardSubmit;
+  newCardPopup.renderLoading(true);
+
+  api
+    .addCard({ name: title, link: url })
+    .then((cardData) => {
+      const cardElement = createCard(cardData);
+      cardSection.addItem(cardElement);
+      newCardPopup.close();
+      addCardForm.reset();
+      addCardFormValidator.enableValidation();
```

Comment:
`enableValidation` should be called only 1 time to enable validation. You should not do that when you open/close a popup because you add more and more the same listeners to the same forms. This may cause a `memory leak`


You should use `addCardFormValidator.disableSubmitButton();` here. Pay attantion, that you need to remove the low dash to use it in `index.js`.